src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -202,20 +233,72 @@ DatabaseType.PostgreSQL or
             }
 
             List<IDefinitionNode> nodes = new(objectTypes.Values);
-            return (new DocumentNode(nodes.Concat(inputObjects.Values).ToImmutableList()), inputObjects);
+            return new DocumentNode(nodes);
         }
 
-        private (DocumentNode, Dictionary<string, InputObjectTypeDefinitionNode>) GenerateCosmosGraphQLObjects()
+        /// <summary>
+        /// Generates the ObjectTypeDefinitionNodes and InputObjectTypeDefinitionNodes as part of GraphQL Schema generation for cosmos db.
+        /// Each datasource in cosmos has a root file provided which is used to generate the schema. 
+        /// </summary>
+        /// <param name="dataSourceNames">Hashset of datasourceNames to generate cosmos objects.</param>
+        /// <returns></returns>
+        private DocumentNode GenerateCosmosGraphQLObjects(HashSet<string> dataSourceNames, Dictionary<string, InputObjectTypeDefinitionNode> inputObjects)
         {
-            Dictionary<string, InputObjectTypeDefinitionNode> inputObjects = new();
-            DocumentNode root = ((CosmosSqlMetadataProvider)_sqlMetadataProvider).GraphQLSchemaRoot;
+            DocumentNode? root = null;
+
+            if (dataSourceNames.Count() == 0)
+            {
+                return new DocumentNode(new List<IDefinitionNode>());
+            }
+
+            foreach (string dataSourceName in dataSourceNames)
+            {
+                ISqlMetadataProvider metadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                DocumentNode currentNode = ((CosmosSqlMetadataProvider)metadataProvider).GraphQLSchemaRoot;
+                root = root is null ? currentNode : root.WithDefinitions(root.Definitions.Concat(currentNode.Definitions).ToImmutableList());
+            }
 
-            IEnumerable<ObjectTypeDefinitionNode> objectNodes = root.Definitions.Where(d => d is ObjectTypeDefinitionNode).Cast<ObjectTypeDefinitionNode>();
+            IEnumerable<ObjectTypeDefinitionNode> objectNodes = root!.Definitions.Where(d => d is ObjectTypeDefinitionNode).Cast<ObjectTypeDefinitionNode>();
             foreach (ObjectTypeDefinitionNode node in objectNodes)
             {
                 InputTypeBuilder.GenerateInputTypesForObjectType(node, inputObjects);
             }
 
+            return root;
+        }
+
+        public (DocumentNode, Dictionary<string, InputObjectTypeDefinitionNode>) GenerateGraphQLObjects()
```

Comment:
It looks like this method can be private. No external callers besides a public method in this class, and then a test class.
To mitigate the test usage, you can use `internals visible to`.. examples: https://github.com/search?q=repo%3AAzure%2Fdata-api-builder+internals+visible&type=code

Reply:
This is discussed offline.

Reply:
Keeping public.